### PR TITLE
Update and document experimental features

### PR DESCRIPTION
**Pull request by writer-jill**
Remove the experimental flag for features that are no longer experimental.
Add a doc to list current experimental features.
This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: docs/development/experimental-features.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## Kubernetes
+
+- [Kubernetes](../development/extensions-core/kubernetes.md)
+
+## Segment locking
+
+- [Configuration reference](../configuration/index.md)
+- [Task reference](../ingestion/tasks.md)
+- [Design](../design/architecture.md)
```
Comment (abhishekagarwal87) [inline docs/development/experimental-features.md:53]: this and configuration reference page is not needed IMO.

File: docs/development/experimental-features.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- [Aggregations](../querying/aggregations.md)
+
+## Front coding
+
+- [Ingestion spec reference](../ingestion/ingestion-spec.md#front-coding)
+
+## Other configuration properties
+
+- [Configuration reference](../configuration/index.md)
+   - Experimental task runner `httpRemote`
```
Comment (abhishekagarwal87) [inline docs/development/experimental-features.md:70]: httpRemote is not considered experimental. I have seen it used in prod systems already.

File: docs/development/experimental-features.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+## Nested columns
+
+- [Nested columns](../querying/nested-columns.md)
+
+## Indexer process
+
+- [Indexer process](../design/indexer.md)
+- [Configuration reference](../configuration/index.md)
+- [Processes and servers](../design/processes.md)
```
[the remark this answers is not in the document]
Comment (writer-jill) [reply docs/development/experimental-features.md:43]: Updated to reference specific headings.

File: docs/development/experimental-features.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## Segment locking
+
+- [Configuration reference](../configuration/index.md)
+- [Task reference](../ingestion/tasks.md)
+- [Design](../design/architecture.md)
+
+## Materialized view
+
+- [Materialized view](../development/extensions-contrib/materialized-view.md)
```
[the remark this answers is not in the document]
Comment (writer-jill) [reply docs/development/experimental-features.md:57]: @abhishekagarwal87 do you want this entire section of docs removed? There's a whole structure of docs underneath extensions-contrib. See https://druid.apache.org/docs/latest/development/extensions-contrib/

File: docs/development/experimental-features.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+## Moments sketch
+
+- [Aggregations](../querying/aggregations.md)
```
[the remark this answers is not in the document]
Comment (writer-jill) [reply docs/development/experimental-features.md:61]: Updated.

File: docs/development/experimental-features.md (hunk 2 above)
Comment (writer-jill) [reply docs/development/experimental-features.md:70]: Yes, this is no longer marked as experimental in the config doc. Removed.

File: docs/development/experimental-features.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (abhishekagarwal87) [reply docs/development/experimental-features.md:57]: that section of docs is not exposed through the doc navigation pane. I think we can keep that as it is. There are many others contrib extensions that are experimental.

File: docs/development/experimental-features.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- [SQL-based ingestion and multi-stage query task API](../multi-stage-query/api.md)
+
+## Nested columns
+
+- [Nested columns](../querying/nested-columns.md)
+
+## Indexer process
+
+- [Indexer process](../design/indexer.md)
+- [Configuration reference](../configuration/index.md#overlord-operations)
```
[the remark this answers is not in the document]
Comment (writer-jill) [reply docs/development/experimental-features.md:42]: Updated.
http reference is already removed - see configuration/index.md in the this PR.

File: docs/development/experimental-features.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (writer-jill) [reply docs/development/experimental-features.md:57]: Removed reference to the materialized view as "experimental"

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**